packages/radix-vue/src/Accordion/AccordionRoot.vue

```diff
@@ -1,34 +1,107 @@
 <script lang="ts">
-import type { AccordionMultipleProps } from "./AccordionImplMultiple.vue";
-import type { AccordionSingleProps } from "./AccordionImplSingle.vue";
+import type { DataOrientation, Direction, Type } from "@/shared/types";
+import type { ComputedRef, InjectionKey, Ref } from "vue";
 
-export type AccordionRootProps = AccordionSingleProps | AccordionMultipleProps;
+export type AccordionRootProps = {
+  asChild?: boolean;
+
+  /**
+   * Determines whether one or multiple items can be opened at the same time.
+   */
+  type: Type;
+
+  /**
+   * The controlled value of the item to expand when type is "single" or the controlled values of the items to expand when type is "multiple".
+   */
+  modelValue?: string | string[];
+
+  /**
+   * The default value of the item to expand when type is "single" or the default values of the items to expand when type is "multiple".
+   * Use when you do not need to control the state of the item(s).
+   */
+  defaultValue?: string | string[];
+
+  /**
+   * When type is "single", allows closing content when clicking trigger for an open item.
+   * When type is "multiple", this prop has no effect.
+   *
+   * @default false
+   */
+  collapsible?: boolean;
+
+  /**
+   * When true, prevents the user from interacting with the accordion and all its items
+   *
+   * @default false
+   */
+  disabled?: boolean;
+
+  /**
+   * The reading direction of the accordion when applicable. If omitted, assumes LTR (left-to-right) reading mode.
+   *
+   * @default "ltr"
+   */
+  dir?: Direction;
+
+  /**
+   * The orientation of the accordion.
+   *
+   * @default "vertical"
+   */
+  orientation?: DataOrientation;
+};
+
+export const ACCORDION_INJECTION_KEY =
+  Symbol() as InjectionKey<AccordionProvideValue>;
+
+export type AccordionProvideValue = {
+  disabled?: AccordionRootProps["disabled"];
+  direction: AccordionRootProps["dir"];
+  orientation: AccordionRootProps["orientation"];
+  parentElement: Ref<HTMLElement | undefined>;
+  changeModelValue(value: string): void;
+  isSingle: ComputedRef<boolean>;
+  modelValue: Ref<string | undefined | string[]>;
+  collapsible: boolean;
+};
 </script>
 
 <script setup lang="ts">
-import AccordionImplMultiple from "./AccordionImplMultiple.vue";
-import AccordionImplSingle from "./AccordionImplSingle.vue";
+import { PrimitiveDiv, usePrimitiveElement } from "@/Primitive";
+import { useSingleOrMultipleValue } from "@/shared/useSingleOrMultipleValue";
+import { computed, provide } from "vue";
 
-const props = defineProps<AccordionRootProps>();
+const props = withDefaults(defineProps<AccordionRootProps>(), {
+  asChild: false,
+  dir: "ltr",
+  disabled: false,
+  orientation: "vertical",
+  collapsible: false,
+});
 
-const emit = defineEmits<{
+const emits = defineEmits<{
   (e: "update:modelValue", value: string | string[] | undefined): void;
 }>();
+
+const { modelValue, changeModelValue } = useSingleOrMultipleValue(props, emits);
```

Comment:
I think you missed out passing `defaultValue` here.

causing the Accordion not open by default in uncontrollable state


```
    <AccordionRoot
        class="w-[300px] rounded-md bg-[--line-color] shadow-lg"
        type="single"
        default-value="item-3"
      >
```
![image](https://github.com/radix-vue/radix-vue/assets/59365435/192a812e-93a9-4cb2-ad5c-966519333d39)

Reply:
I will get the value directly from the props. I think is cleaner